### PR TITLE
Consolidate

### PR DESCRIPTION
**Pull request by 2justinmorgan**
I have more to do

### CODE REVIEW

File: node.go [hunk 1]
```diff
@@ -80,6 +80,33 @@ func (node *Node) run(db *DB) {
...
+	a := make([]Data,len(m))
+	for _,val := range m {
+		a[i] = val
+		i += 1
+	}
+	return a
+}
+
+func consolidateDataVals(vals []Data) []Data {
+	m := make(map[int]Data)
```
Comment (2justinmorgan) [inline node.go:94]: Returns an array of Data vals with the highest clock values associated to their respective nodes

File: node.go [hunk 2]
```diff
@@ -80,6 +80,33 @@ func (node *Node) run(db *DB) {
...
+	i := 0
+	a := make([]Data,len(m))
+	for _,val := range m {
+		a[i] = val
+		i += 1
+	}
+	return a
+}
+
+func consolidateDataVals(vals []Data) []Data {
```
Comment (2justinmorgan) [inline node.go:93]: Returns an array of Data vals with the highest clock values associated to their respective nodes

File: node.go [hunk 3]
```diff
@@ -110,7 +137,7 @@ func (node *Node) handleGet(request Request) {
 			responses += 1
 		}
 		//TODO: consolidate vals
-		node.getReturn <- vals
+		node.getReturn <- consolidateDataVals(vals)
```
Comment (2justinmorgan) [inline node.go:140]: Returns an array of Data vals with the highest clock values associated to their respective nodes

**Comment by awhite37** [comment]
you might already be aware because you said you had more to do but this only works for vector clocks with a single clock value entry. It does for work the example run though, so I can use it to finish the results section of the write up

**Pull request merged**